opencti-platform/opencti-graphql/src/utils/exclusionLists.ts

```diff
@@ -0,0 +1,107 @@
+import { MAX_EVENT_LOOP_PROCESSING_TIME } from '../database/utils';
+import { FunctionalError } from '../config/errors';
+import { type ExclusionListProperties } from './exclusionListsTypes';
+
+export const getIsRange = (value: string) => value.indexOf('/') !== -1;
+
+export const checkIpAddrType = (ipAddr: string) => {
+  const isIpv4 = ipAddr.split('.').length === 4;
+  const isIpv6 = ipAddr.indexOf(':') !== -1;
+  return { isIpv4, isIpv6 };
+};
+
+export const convertIpv6ToBinary = (ipv6: string, isRange?: boolean, range?: number) => {
+  let test = ipv6.split(':');
+  const emptyFieldIndex = test.indexOf('');
+
+  if (isRange && range) {
+    const binary = ipv6
+      .split(':')
+      .map((t) => (t === '' ? '0' : t))
+      .map((hex) => (parseInt(hex, 16).toString(2)).padStart(16, '0'))
+      .join('');
+    return binary.slice(0, range);
+  }
+
+  if (emptyFieldIndex !== -1 && test.length < 8) {
+    test = [
+      ...test.slice(0, emptyFieldIndex + 1),
+      ...Array(8 - test.length).fill('0'),
+      ...test.slice(emptyFieldIndex + 1)
+    ];
+  }
+
+  return test
+    .map((t) => (t === '' ? '0' : t))
+    .map((hex) => (parseInt(hex, 16).toString(2)).padStart(16, '0'))
+    .join('');
+};
+
+export const convertIpv4ToBinary = (ipv4: string, isRange?: boolean, range?: number) => {
+  const binary = ipv4.split('.').map((ip) => (parseInt(ip, 10).toString(2)).padStart(8, '0')).join('');
+  if (isRange && range) {
+    return binary.slice(0, range);
+  }
+  return binary;
+};
+
+const throwExclusionListError = (value: string, listName: string) => {
+  throw FunctionalError(`Indicator creation failed, this pattern (${value}) is contained on an exclusion list (${listName})`, { value });
```

Comment:
I agree. I think it should be refactored as so:
- checkExclusionList and checkIpAddressLists should be changed to return either the id/name of the exclusion list matched, or null if nothing matches
- throwExclusionListError should be changed to a applyExclusionList method, taking an indicator instance and a list name in parameter. For now, this method will throw an exception just the same way, but it will be used in the future when more actions are implemented